src/content/en/updates/2018/07/page-lifecycle-api.md

```diff
@@ -447,12 +447,19 @@ that pertain to lifecycle and lists what states they may transition to and from.
       point.</p>
       <aside class="warning">
         <strong>Warning:</strong>
-        The <code>beforeunload</code> event should only be used to alert the
-        user of unsaved changes. Once those changes are saved, the event should
-        be removed. It should never be added unconditionally to the
-        page, as doing so can hurt performance in some cases. See the
-        <a href="#legacy-lifecycle-apis-to-avoid">legacy APIs section</a>
-        for details.
+        The <code>beforeunload</code> event should only be used in two cases:
```

Comment:
I think I'd prefer to only mention this in the "Managing cross-browsers differences" section. There are lots of subtle compat issues with the advice here, and I think it would be distracting to list it all inline rather than consolidated in a single section.

Reply:
Maybe adapt the comment text and say something like:

"Safari does not reliably fire the `pagehide` or `visibilitychange` events when closing a tab (webkit bugs: [151610](https://bugs.webkit.org/show_bug.cgi?id=151610) and [151234](https://bugs.webkit.org/show_bug.cgi?id=151234)), so in Safari you may need to *also* listen to the `beforeunload` event in order to detect a change to the hidden state. But since the `beforeunload` event can be cancelled, you need [to wait until after the event has finished propagating](https://github.com/GoogleChromeLabs/page-lifecycle/blob/0.1.1/src/Lifecycle.mjs#L156-L172) to know if the state has changed to hidden. **Important:** using the `beforeunload` event this way should *only* be done in Safari, as using this event in other browsers can hurt performance. See the [legacy APIs section](#legacy-lifecycle-apis-to-avoid) for details."

Reply:
Done.